llap-server/src/java/org/apache/hadoop/hive/llap/io/encoded/OrcEncodedDataReader.java

```diff
@@ -276,11 +275,20 @@ public void unpause() {
 
   @Override
   protected Void callInternal() throws IOException, InterruptedException {
-    return ugi.doAs(new PrivilegedExceptionAction<Void>() {
-      @Override
-      public Void run() throws Exception {
-        return performDataRead();
+    return ugi.doAs((PrivilegedExceptionAction<Void>) () -> {
+      long startTime = counters.startTimeCounter();
+      try {
+        performDataRead();
+        consumer.setDone();
+      } catch (Throwable t) {
+        consumer.setError(t);
+        trace.dumpLog(LOG);
+      } finally {
+        cleanupReaders();
+        recordReaderTime(startTime);
```

Comment:
should this be the 1st step?

Reply:
I do not know, that is confusing for me too. If you look at previous code, in some case recordReaderTime is called before cleanupReaders (Ex: end of function and handleError) and in some cases its called after (Ex: processStop). I do not think it was intentional. I can do anything which seems correct, cleanupReader is closing IO objects, whether that part of IO or not is subjective. I can do it the way you feel strongly.

Reply:
if that is supposed to profile the read time, it should probably be before the cleanup

Reply:
Done